### PR TITLE
clear the route cache for ext authz

### PR DESCRIPTION
**Pull request by aryan16**
**Please provide a description of this PR:**

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/security/authz/builder/extauthz.go [hunk 1]
```diff
@@ -289,6 +289,7 @@ func generateHTTPConfig(hostname, cluster string, status *envoytypev3.HttpStatus
 		},
 		FilterEnabledMetadata: generateFilterMatcher(wellknown.HTTPRoleBasedAccessControl),
 		WithRequestBody:       withBodyRequest(config.IncludeRequestBodyInCheck),
+		ClearRouteCache:       true,
```
Comment (howardjohn) [inline pilot/pkg/security/authz/builder/extauthz.go:292]: do we need this conditional? https://github.com/istio/istio/blob/24c6d2151bd1703b76047da97f26f1a8df2edaf2/pilot/pkg/security/authn/v1beta1/policy_applier.go#L163

Comment (aryan16) [reply pilot/pkg/security/authz/builder/extauthz.go:292]: that one is for reqAuthn, I don't think it has anything to do with extAuthz.

Comment (howardjohn) [reply pilot/pkg/security/authz/builder/extauthz.go:292]: Doesn't the same apply though? we don't do routing for inbound trafffic for sidecars. Only gw?

Comment (aryan16) [reply pilot/pkg/security/authz/builder/extauthz.go:292]: yeah, so you mean to say we should follow same for external Authz as well? 
Eg. Don't need to clear the cache when it's for sidecars and only clear it when it is for gw

Comment (aryan16) [reply pilot/pkg/security/authz/builder/extauthz.go:292]: I agree, will update this PR to ignore cache clearing for sidecars

Comment (howardjohn) [reply pilot/pkg/security/authz/builder/extauthz.go:292]: still working on ^?

**Comment by hzxuzhonghu** [comment]
please describe a little bit more about context

**Pull request closed without merging**